iOS/Flows/Conversation/Cells/TimeMachineCollectionViewLayout.swift

```diff
@@ -14,13 +14,25 @@ protocol TimeMachineCollectionViewLayoutDataSource: AnyObject {
     func getMessage(at indexPath: IndexPath) -> Messageable?
 }
 
+class TimeMachineCollectionViewLayoutInvalidationContext: UICollectionViewLayoutInvalidationContext {
+    /// If true, the z ranges for all the items should be recalculated.
+    var shouldRecalculateZRanges = true
+}
+
```

Comment:
Running all the sorting math every time the offset changed was not performant with large sets of messages.